p2p-media-loader-hlsjs/lib/hlsjs-loader.ts

```diff
@@ -80,7 +101,15 @@ export class HlsJsLoader {
         this.stats.trequest = now - DEFAULT_DOWNLOAD_LATENCY - downloadTime;
         this.stats.tfirst = now - downloadTime;
         this.stats.tload = now;
+        this.stats.loading = {
+            first: this.stats.tfirst,
+            start: this.stats.trequest,
+            end: this.stats.tload
+        };
         this.stats.loaded = content.byteLength;
+        this.stats.total = content.byteLength;
+
+        callbacks.onProgress(this.stats, context, content);
```

Comment:
```suggestion
        if(callbacks.onProgress) {
            callbacks.onProgress(this.stats, context, content);
        }
```
AES key file does not require onProgress callback so adding this check will prevent ```onProgress is not a function``` error